script.js

```diff
@@ -1,64 +1,207 @@
+            insereX(2,1);
+            marca_posicao[2][1] = 'X'
+        } else if (marca_posicao[0][0] == 'O' && marca_posicao[1][0] == 'O' && marca_posicao[2][0] == false ){
+            insereX(2,0);
+            marca_posicao[2][0] = 'X'
+        } else if (marca_posicao[0][0] == 'O' && marca_posicao[2][0] == 'O' && marca_posicao[1][0] == false ){
+            insereX(1,0);
+            marca_posicao[1][0] = 'X'
+        } else if (marca_posicao[1][0] == 'O' && marca_posicao[2][0] == 'O' && marca_posicao[0][0] == false ){
+            insereX(0,0);
+            marca_posicao[0][0] = 'X'
+        } else if (marca_posicao[0][1] == 'O' && marca_posicao[1][1] == 'O' && marca_posicao[2][1] == false ){
+            insereX(2,1);
+            marca_posicao[2][1] = 'X'
+        } else if (marca_posicao[0][1] == 'O' && marca_posicao[2][1] == 'O' && marca_posicao[1][1] == false ){
+            insereX(1,1);
+            marca_posicao[1][1] = 'X'
+        } else if (marca_posicao[1][1] == 'O' && marca_posicao[2][1] == 'O' && marca_posicao[0][1] == false ){
+            insereX(0,1);
+            marca_posicao[0][1] = 'X'
+        } else if (marca_posicao[1][2] == 'O' && marca_posicao[2][2] == 'O' && marca_posicao[0][2] == false ){
+            insereX(0,2);
+            marca_posicao[0][2] = 'X'
+        } else if (marca_posicao[0][2] == 'O' && marca_posicao[2][2] == 'O' && marca_posicao[1][2] == false ){
+            insereX(1,2);
+            marca_posicao[1][2] = 'X'
+        } else if (marca_posicao[0][2] == 'O' && marca_posicao[1][2] == 'O' && marca_posicao[2][2] == false ){
+            insereX(2,2);
+            marca_posicao[2][2] = 'X'
+
+        } else if (marca_posicao[0][0] == 'O' && marca_posicao[1][1] == 'O' && marca_posicao[2][2] == false ){
+            insereX(2,2);
+            marca_posicao[2][2] = 'X'
+        } else if (marca_posicao[0][0] == 'O' && marca_posicao[2][2] == 'O' && marca_posicao[1][1] == false ){
+            insereX(1,1);
+            marca_posicao[1][1] = 'X'
+        } else if (marca_posicao[1][1] == 'O' && marca_posicao[2][2] == 'O' && marca_posicao[0][0] == false ){
+            insereX(0,0);
+            marca_posicao[1][1] = 'X'
+
+        } else if (marca_posicao[2][0] == 'O' && marca_posicao[1][1] == 'O' && marca_posicao[0][2] == false ){
+            insereX(0,2);
+            marca_posicao[0][2] = 'X'
+        } else if (marca_posicao[2][0] == 'O' && marca_posicao[0][2] == 'O' && marca_posicao[1][1] == false ){
+            insereX(1,1);
+            marca_posicao[1][1] = 'X'
+        } else if (marca_posicao[1][1] == 'O' && marca_posicao[0][2] == 'O' && marca_posicao[2][0] == false ){
+            insereX(2,0);
+            marca_posicao[2][0] = 'X'
+        }                      
+
+        else if (marca_posicao[linhaRandom][colunaRandom] == false) {
+            marca_posicao[linhaRandom][colunaRandom] = 'xis';
+            marca_posicao[linhaRandom][colunaRandom] = 'X';
```

Comment:
What happens here?

First you say that `marca_posicao[linhaRandom][colunaRandom]` gets `xis` and then you just say that `marca_posicao[linhaRandom][colunaRandom]` gets `X`.

So why applying `xis` first? It's being overwritten in the next line.